MekHQ/src/mekhq/gui/PlanetarySystemMapPanel.java

```diff
@@ -0,0 +1,704 @@
+     * @param limit - an integer indicating how large the text can be
+     * @return
+     */
+    private boolean areNamesTooBig(Graphics g, PlanetarySystem system, DateTime when, int limit) {
+        for(Planet p : system.getPlanets()) {
+            String name = p.getName(when);
+            if(g.getFontMetrics().stringWidth(name) > limit) {
+                //try splitting
+                if(name.contains(" ")) {
+                    for (String line : name.split("\\s+")) {
+                        if(g.getFontMetrics().stringWidth(line) > limit) {
+                            return true;
+                        }
+                    }
+                } else if(name.contains("-")) {
+                    for (String line : name.split("-")) {
+                        if(g.getFontMetrics().stringWidth(line) > limit) {
+                            return true;
+                        }
+                    }
+                }else {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+    
+    /**
+     * Draw a string that is centered on x and y. It will wrap text by spaces of hyphens
+     * if too large.
+     * @param g - a <code>Graphics2D</code> device to draw on
+     * @param text - a <code>String</code> of text to be drawn
+     * @param x - an <code>int</code> for the x-axis position
+     * @param y - an <code>int</code> for the y-axis position
+     * @param limit - an <code>int</coce> for how big the text can be
+     */
+    private void drawCenteredString(Graphics2D g, String text, int x, int y, int limit) {
+        FontMetrics metrics = g.getFontMetrics();
+        y = y - (metrics.getHeight() / 2);
+        if(metrics.stringWidth(text) > limit && (text.contains(" ") || text.contains("-"))) {
+            if(text.contains(" ")) {
+                //try spaces first
+                for (String line : text.split("\\s+")) {
+                    g.drawString(line, x- (metrics.stringWidth(line) / 2), y += g.getFontMetrics().getHeight());
+                }
+            } else {
+                //otherwise break on the hyphen
+                String[] lines = text.split("-");
+                for (int i = 0; i < lines.length; i++) {
+                    String line = lines[i];
+                    if(i != (lines.length-1)) {
+                        line = line + "-";
+                    }
+                    g.drawString(line, x- (metrics.stringWidth(line) / 2), y += g.getFontMetrics().getHeight());
+                }
+            }
+        } else {
+            g.drawString(text, x- (metrics.stringWidth(text) / 2), y += g.getFontMetrics().getHeight());
+        }
+    }
+    
+    /**
+     * update and repaint the panel for a new planetary system
+     * @param s - a <code>PlanetarySystem</code> to paint
+     */
+    public void updatePlanetarySystem(PlanetarySystem s) {
+        this.system = s;
+        selectedPlanet = system.getPrimaryPlanetPosition();
```

Comment:
Do we need a null check on `s`?

Reply:
I haved added a null check that just returns without doing anything